scripts/compare-metrics.sh

```
@@ -0,0 +1,383 @@
+extract_value() {
+    local response="$1"
+    local default="${2:-0}"
+
+    # Check if query was successful
+    if [[ $(echo "$response" | jq -r '.status' 2>/dev/null) != "success" ]]; then
+        echo "$default"
+        return 1
+    fi
+
+    # Extract first result value
+    local value=$(echo "$response" | jq -r ".data.result[0].value[1] // \"$default\"" 2>/dev/null)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)